src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -113,7 +117,17 @@ public async Task StartAsync(Uri url, TransferFormat transferFormat, Cancellatio
                 var accessToken = await _accessTokenProvider();
                 if (!string.IsNullOrEmpty(accessToken))
                 {
-                    _webSocket.Options.SetRequestHeader("Authorization", $"Bearer {accessToken}");
+                    // We can't use request headers in the browser, so instead append the token as a query string in that case
+                    if (_isRunningInBrowser)
+                    {
+                        var accessTokenEncoded = UrlEncoder.Default.Encode(accessToken);
```

Comment:
Would be nice to test this code path. Maybe we can make `_isRunningInBrowser` internal and modify it in tests to be true.

Reply:
Writing a test for this is probably something I'll need help with - do you know of a test I can use as an example of some similar scenario?

Reply:
Probably something similar to https://github.com/dotnet/aspnetcore/blob/20f6d6582896344cad45adb82cccc445d1e8e762/src/SignalR/server/SignalR/test/WebSocketsTransportTests.cs#L68